src/com/reco1l/osu/data/Scores.kt

```diff
@@ -0,0 +1,197 @@
+    /**
+     * The maximum combo.
+     */
+    val maxCombo: Int,
+
+    /**
+     * The mark.
+     */
+    val mark: String,
+
+    /**
+     * The number of 300k hits.
+     */
+    val hit300k: Int,
+
+    /**
+     * The number of 300 hits.
+     */
+    val hit300: Int,
+
+    /**
+     * The number of 100k hits.
+     */
+    val hit100k: Int,
+
+    /**
+     * The number of 100 hits.
+     */
+    val hit100: Int,
+
+    /**
+     * The number of 50 hits.
+     */
+    val hit50: Int,
+
+    /**
+     * The number of misses.
+     */
+    val misses: Int,
+
+    /**
+     * The accuracy.
+     */
+    val accuracy: Float,
+
+    /**
+     * The score date.
+     */
+    val time: Long,
+
+    /**
+     * Whether the score is perfect.
+     */
+    val isPerfect: Boolean
+
+) {
+
+    fun toJSON() = JSONObject().apply {
+
+        // The keys doesn't correspond to the table columns in order to keep compatibility with the old replays.
+
+        put("id", id)
+        put("filename", beatmapFilename)
+        put("playername", playerName)
+        put("replayfile", replayPath)
+        put("mod", mods)
+        put("score", score)
+        put("combo", maxCombo)
+        put("mark", mark)
+        put("h300k", hit300k)
+        put("h300", hit300)
+        put("h100k", hit100k)
+        put("h100", hit100)
+        put("h50", hit50)
+        put("misses", misses)
+        put("accuracy", accuracy)
+        put("time", time)
+        put("isPerfect", if (isPerfect) 1 else 0)
+
+    }
+
+    fun toStatisticV2() = StatisticV2().also {
+
+        it.playerName = playerName
+        it.replayName = replayPath
+        it.setModFromString(mods)
+        it.setForcedScore(score)
+        it.maxCombo = maxCombo
+        it.mark = mark
+        it.hit300k = hit300k
+        it.hit300 = hit300
+        it.hit100k = hit100k
+        it.hit100 = hit100
+        it.hit50 = hit50
+        it.misses = misses
+        it.accuracy = accuracy
+        it.time = time
+        it.isPerfect = isPerfect
+
+    }
+
+
+}
+
+fun ScoreInfo(json: JSONObject) = ScoreInfo(
+
+    // The keys doesn't correspond to the table columns in order to keep compatibility with the old replays.
+
+    id = json.optLong("id", 0),
+    beatmapFilename = json.getString("filename"),
+    playerName = json.getString("playername"),
+    replayPath = json.getString("replayfile"),
+    mods = json.getString("mod"),
+    score = json.getInt("score"),
+    maxCombo = json.getInt("combo"),
+    mark = json.getString("mark"),
+    hit300k = json.getInt("h300k"),
+    hit300 = json.getInt("h300"),
+    hit100k = json.getInt("h100k"),
+    hit100 = json.getInt("h100"),
+    hit50 = json.getInt("h50"),
+    misses = json.getInt("misses"),
+    accuracy = json.getDouble("accuracy").toFloat(),
+    time = json.getLong("time"),
+    isPerfect = json.getInt("isPerfect") == 1
+)
+
+
+@Dao
+interface IScoreInfoDAO {
+
+    @Query("SELECT * FROM ScoreInfo WHERE beatmapFilename = :beatmapFilename")
```

Comment:
`beatmapFilename` should be indexed to prevent a sequential scan in case the table is large enough. This will also benefit `getBestMark` and `getBestScoreId` queries.

Reply:
[742592e](https://github.com/osudroid/osu-droid/pull/390/commits/742592e67008b14badf7568da20c9c72a4896a8c)